data/scripts.js

```diff
@@ -2428,13 +2444,13 @@ exports.BattleScripts = {
 					if (hasMove['bravebird']) rejected = true;
 					break;
 				case 'solarbeam':
-					if ((!hasMove['sunnyday'] && template.species !== 'Ninetales') || hasMove['gigadrain'] || hasMove['leafstorm']) rejected = true;
+					if ((!hasMove['sunnyday'] && !hasAbility['Drought']) || hasMove['gigadrain'] || hasMove['leafstorm']) rejected = true;
 					break;
 				case 'gigadrain':
-					if ((!setupType && hasMove['leafstorm']) || hasMove['petaldance']) rejected = true;
+					if ((!setupType && hasAbility['Contrary'] && hasMove['leafstorm']) || hasMove['petaldance']) rejected = true;
```

Comment:
This doesn't make sense. It should be rejected for Simisage, Celebi, etc... and not only Serperior.